src/overlays/actors/ovl_En_Fsn/z_en_fsn.c

```diff
@@ -21,130 +90,1557 @@ const ActorInit En_Fsn_InitVars = {
+            globalCtx->msgCtx.unk11F22 = 0x43;
+            globalCtx->msgCtx.unk12023 = 4;
+            this->actionFunc = EnFsn_IdleBackroom;
+        } else if (ENFSN_SHOULD_GIVE_ITEM(this)) {
+            ENFSN_UNSET_SHOULD_GIVE_ITEM(this);
+            globalCtx->msgCtx.unk11F22 = 0x43;
+            globalCtx->msgCtx.unk12023 = 4;
+            this->actionFunc = EnFsn_GiveItem;
+        } else {
+            EnFsn_HandleConversationBackroom(this, globalCtx);
+        }
+    }
+}
+
+void EnFsn_GetCutscenes(EnFsn* this) {
+    this->lookToShopkeeperCutscene = this->actor.cutscene;
+    this->lookToShelfCutscene = ActorCutscene_GetAdditionalCutscene(this->lookToShopkeeperCutscene);
+    this->lookToShopkeeperFromShelfCutscene = ActorCutscene_GetAdditionalCutscene(this->lookToShelfCutscene);
+    this->lookToShopkeeperBuyingCutscene = ActorCutscene_GetAdditionalCutscene(this->lookToShopkeeperFromShelfCutscene);
+}
+
+void EnFsn_Blink(EnFsn* this) {
+    s16 decr = this->blinkTimer - 1;
+
+    if (decr >= 3) {
+        this->eyeTextureIdx = 0;
+        this->blinkTimer = decr;
+    } else if (decr == 0) {
+        this->eyeTextureIdx = 2;
+        this->blinkTimer = (s32)(Rand_ZeroOne() * 60.0f) + 20;
+    } else {
+        this->eyeTextureIdx = 1;
+        this->blinkTimer = decr;
+    }
+}
+
+void EnFsn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFsn* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 20.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06013320, &D_06012C34, this->limbDrawTable,
+                     this->transitionDrawTable, 19);
+    if (ENFSN_IS_SHOP(&this->actor)) {
+        this->actor.shape.rot.y = BINANG_ROT180(this->actor.shape.rot.y);
+        this->actor.flags &= ~1;
+        EnFsn_GetCutscenes(this);
+        EnFsn_InitShop(this, globalCtx);
+    } else {
+        if ((gSaveContext.weekEventReg[0x21] & 8) || (gSaveContext.weekEventReg[0x4F] & 0x40)) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        this->blinkTimer = 20;
+        this->eyeTextureIdx = 0;
+        this->actor.flags |= 1;
+        this->actor.targetMode = 0;
+        this->animationIdx = 0;
+        func_8013BC6C(&this->skelAnime, sAnimations, this->animationIdx);
+        this->actionFunc = EnFsn_IdleBackroom;
+    }
+}
+
+void EnFsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFsn* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnFsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFsn* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_800E9250(globalCtx, &this->actor, &this->unk274, &this->unk27A, this->actor.focus.pos);
```

Comment:
Anyone got any ideas what unk274 and unk27A might be? unk27A doesn't get used anywhere else, but unk274.y does get used once later on with limb index 16 (head?).

Reply:
This may give you some ideas.
<https://github.com/zeldaret/mm/blob/5453731f4e2cadabc518b76ebb1ea9d9d6f366b7/src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c#L644>